pkg/util/bootstraptoken/util.go

```diff
@@ -0,0 +1,44 @@
+package bootstraptoken
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstraptokenapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraptokenutil "k8s.io/cluster-bootstrap/token/util"
+)
+
+func GenerateBootstrapToken() (string, *corev1.Secret, error) {
+
+	token, err := bootstraptokenutil.GenerateBootstrapToken()
+	if err != nil {
+		return "", nil, err
+	}
+	substrs := bootstraptokenutil.BootstrapTokenRegexp.FindStringSubmatch(token)
+	// TODO: Add a constant for the 3 value here, and explain better why it's needed (other than because how the regexp parsin works)
```

Comment:
or not explain it and use https://github.com/kubernetes/cluster-bootstrap/blob/master/util/tokens/tokens.go